pkg/verifier/notation/notation.go

```diff
@@ -181,7 +187,41 @@ func getVerifierService(conf *NotationPluginVerifierConfig, pluginDirectory stri
 	if err != nil {
 		return nil, err
 	}
-	verifier, err := notationVerifier.New(&conf.TrustPolicyDoc, store, NewRatifyPluginManager(pluginDirectory))
+
+	// revocation check using corecrl from notation-core-go and crl from notation-go
+	// This is the implementation for revocation check from notation cli to support crl and cache configurations
+	// removed timeout
+	crlFetcher, err := corecrl.NewHTTPFetcher(&http.Client{}) // Todo: replace with crlprovider notation implementation function
+	if err != nil {
+		return nil, err
+	}
+	cacheRoot, err := dir.CacheFS().SysPath(dir.PathCRLCache)
+	if err != nil {
+		return nil, err
+	}
+	crlFetcher.Cache, err = crl.NewFileCache(cacheRoot) // Todo: replace with crlprovider notation implementation function
+	if err != nil {
+		return nil, err
+	}
+	revocationCodeSigningValidator, err := revocation.NewWithOptions(revocation.Options{
+		CRLFetcher:       crlFetcher,
+		CertChainPurpose: purpose.CodeSigning,
+	})
+	if err != nil {
+		return nil, err
+	}
+	revocationTimestampingValidator, err := revocation.NewWithOptions(revocation.Options{
+		CRLFetcher:       crlFetcher,
```

Comment:
qq: I saw notation CLI would configure an ocsp http client, wonder if the OCSP revocation check will work without explicitely setting it?

Reply:
As defined in notation-core-go there is a default value for performance concern we can add the http client to cancel the default timeout.
```
	// OCSPHTTPClient is the HTTP client for OCSP request. If not provided,
	// a default *http.Client with timeout of 2 seconds will be used.
	// OPTIONAL.
	OCSPHTTPClient *http.Client
```